examples/mml_parser/mml_parser_main.c

```diff
@@ -0,0 +1,197 @@
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: print_parse_result
+ ****************************************************************************/
+
+int print_parse_result(int ret_code, FAR struct mml_result_s *result)
+{
+  int passed_tick = 0;
+
+  switch (ret_code)
+    {
+      case MML_TYPE_NOTE:
+        printf("Note(%2d) : length = %d\n",
+            result->note_idx[0], result->length);
+        passed_tick += result->length;
+        break;
+      case MML_TYPE_REST:
+        printf("Rest     : length = %d\n", result->length);
+        passed_tick += result->length;
+        break;
+      case MML_TYPE_TEMPO:
+        printf("Tempo    : length = %d\n", result->length);
+        break;
+      case MML_TYPE_LENGTH:
+        printf("Length   : length = %d\n", result->length);
+        break;
+      case MML_TYPE_OCTAVE:
+        printf("Octave   : length = %d\n", result->length);
+        break;
+      case MML_TYPE_TUPLETSTART:
+        printf("Tuplet B : length = %d\n", result->length);
+        break;
+      case MML_TYPE_TUPLETDONE:
+        printf("Tuplet D :\n");
+        break;
+      case MML_TYPE_VOLUME:
+        printf("Volume   : length = %d\n", result->length);
+        break;
+      case MML_TYPE_CHORD:
+        printf("Chord    : length = %d, chord_num = %d\n",
+            result->length, result->chord_notes);
+        printf("         : Notes ");
+        {
+          int i;
+          for (i = 0; i < result->chord_notes; i++)
+            {
+              printf("%d ", result->note_idx[i]);
+            }
+
+          printf("\n");
+        }
+
+        passed_tick += result->length;
+        break;
+    }
+
+  return passed_tick;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: main
+ ****************************************************************************/
+
+int main(void)
+{
+  int i;
+  int ret;
+  int total_tick;
+
+  FAR char *score;
+  struct music_macro_lang_s mml;
+  struct mml_result_s result;
+
+  for (i = 0; i < TEST_SCORES_NUM; i++)
+    {
+      init_mml(&mml, 48000, 1, 0, 4);
+
+      score = (char *)test_scores[i];
+      printf("===================================\n");
+      printf("Test Score :\n%s\n\n", score);
+
+      total_tick = 0;
+      while ((ret = parse_mml(&mml, &score, &result)) > 0)
+        {
+          if (ret < 0)
+            {
+              printf("\nret = %d\n", ret);
+              printf("Error was occured below:\n");
+              printf("%s\n", score);
+              break;
+            }
+          else
```

Comment:
I think that this condition is a dead code as while is checking `ret > 0`, so I do not see how this can be hit.

Reply:
Yes. That is correct.
Should I delete it?
But, I think that the sample code also serves to show how to use the library, so I think it is better to keep this as the error handling code, even if it didn't hit.
What do you think?

Or I can also add a comment like "In case of an error is occurred." in "if" statement.

Reply:
I'm ok to have this condition, but let's move it outside the loop.
```
while ((ret = parse_mml(&mml, &score, &result)) > 0)
  {
...
  }

if (ret < 0)
  {
    printf("\nret = %d\n", ret);
    printf("Error was occured below:\n");
    printf("%s\n", score);
  } 
```
Do you see this reasonable?

Reply:
Oh. Sorry. I had mis-understood.,

Yes, you are correct.
And your code is fine.

I changed it.
Please check.

Thank you.